packages/namekit-react/src/components/PriceSymbol.tsx

```diff
@@ -0,0 +1,88 @@
+import { PriceCurrencyFormat, Currency } from "@namehash/ens-utils";
+import { Tooltip } from "@namehash/nameguard-react";
+
+import EthereumCircleIcon from "./EthereumCircleIcon";
+import { UsdcSymbol } from "./UsdcSymbol";
+import { WethSymbol } from "./WethSymbol";
+import { DaiSymbol } from "./DaiSymbol";
+import React from "react";
+
+export enum PriceDisplaySize {
+  Micro = "text-xs md:text-sm font-normal",
+  Small = "text-sm font-semibold",
+  Medium = "text-xl font-semibold",
+  Large = "text-xl font-semibold md:text-2xl md:font-bold",
+}
+
+export enum PriceSymbolPosition {
+  Left = "mr-1.5",
+  Right = "ml-1.5",
+}
+
+export enum PriceDisplayPosition {
+  Right = "flex inline-flex items-end space-x-2",
+  Bottom = "flex flex-col text-right items-end space-y-1",
+}
+
+export enum PriceSymbology {
+  Acronym = "Acronym",
+  Symbol = "Symbol",
+}
+
+interface PriceSymbolProps {
+  currency: Currency;
+  size: PriceDisplaySize;
+  symbolFillColor?: string;
+  showTooltipDescription: boolean;
+}
+
+const SymbolSize: Record<PriceDisplaySize, string> = {
+  [PriceDisplaySize.Micro]: "w-4",
+  [PriceDisplaySize.Small]: "w-4",
+  [PriceDisplaySize.Medium]: "w-5",
+  [PriceDisplaySize.Large]: "w-5",
+};
+
+export const PriceSymbol = ({
+  size,
+  currency,
+  symbolFillColor,
+  showTooltipDescription,
+}: PriceSymbolProps) => {
+  const symbol =
+    currency === Currency.Usd ? (
```

Comment:
Could we make use of a switch statement here? I prefer to use switch where we want to ensure we have an answer for every possible enum value. with switch the type system can help to raise an error if we miss some case. Just ensure you don't use `default` in the switch. All cases should be explicit.